### ISSUE TITLE
Unable to access a google_dns_managed_zone data source with provider version v3.0.0-beta.1

### ISSUE THREAD

**Issue by bdubertret** [terraform-providers/terraform-provider-google#5001]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Affected Resource(s)

* google_dns_managed_zone

### Terraform Configuration Files

```tf
data "google_dns_managed_zone" "zone" {
    name = "test-zone"
    project = "iscool-dev"
}
```

### Debug Output

https://gist.github.com/bdubertret/549c26910f2bdede1d4377e1318d2348

### Expected Behavior

Terraform should retrieve the managed zone details without error

### Actual Behavior

```
data.google_dns_managed_zone.zone: Refreshing state...

Error: googleapi: Error 404: The 'parameters.managedZone' resource named 'projects/iscool-dev/managedZones/test-zone' does not exist., notFound

  on main.tf line 1, in data "google_dns_managed_zone" "zone":
   1: data "google_dns_managed_zone" "zone" {


```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

This bug is only present in  v3.0.0-beta.1 and seems to be introduced by commit 1b413bc :

https://github.com/terraform-providers/terraform-provider-google/commit/1b413bcb4e704933b16f61207b4f70c0cb6a4dbb#diff-e3486425f56b00142cb9244ebc1c0880

**Comment by migs** [comment]
This seems to be fixed in https://github.com/terraform-providers/terraform-provider-google-beta/pull/1405 - so I guess we just need to wait for it to be brought into the next version

**Comment by slevenick** [comment]
Yep, this was reported here: https://github.com/terraform-providers/terraform-provider-google/issues/4939

And will be fixed in the next release (3.0.0)
